### PR TITLE
Logging driver 920

### PR DESCRIPTION
**Pull request by bermudezmt**
Backports changes that have been merged into vnext-branch:

Please pay attention to /config/containers/logging/local.md as the file has been updated 6 days ago.

### CODE REVIEW

**Comment by bermudezmt** [comment]
@darwin-traver can you please have a look? Make sure we have the latest info especially on the local.md file.

File: config/containers/logging/configure.md [hunk 1]
```diff
@@ -157,9 +150,12 @@ see more options.
 | [`etwlogs`](etwlogs.md)       | Writes log messages as Event Tracing for Windows (ETW) events. Only available on Windows platforms.           |
 | [`gcplogs`](gcplogs.md)       | Writes log messages to Google Cloud Platform (GCP) Logging.                                                   |
 | [`logentries`](logentries.md) | Writes log messages to Rapid7 Logentries.                                                                     |
+| [`local`](local.md)           | Logs are stored in a custom format designed for minimal overhead.                                             |
 
 ## Limitations of logging drivers
 
-The `docker logs` command is not available for drivers other than `json-file`
-and `journald`.
+The `docker logs` command is only available on the following drivers:
```
Comment (darwin-traver) [inline config/containers/logging/configure.md]: Do we want to make a reference note to 'Dual Logging' that will allow 'docker logs' to be used with third-party logging drivers?

File: config/containers/logging/configure.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ unless you configure it to use a different logging driver.
 
 In addition to using the logging drivers included with Docker, you can also
 implement and use [logging driver plugins](/engine/admin/logging/plugins.md).
+Logging driver plugins are available in Docker 17.05 and higher.
```
Comment (thaJeztah) [inline config/containers/logging/configure.md]: Do we need to keep this? 17.05 is obsolete for a long time, so don't think worth mentioning

File: config/containers/logging/json-file.md [hunk 3]
```diff
@@ -27,22 +23,17 @@ configuring Docker using `daemon.json`, see
...
-    "max-size": "10m",
-    "max-file": "3"
+    "max-size": "10m"
   }
 }
 ```
 
-> **Note**: `log-opt` configuration options in the `daemon.json` configuration
-> file must be provided as strings. Boolean and numeric values (such as the value
-> for `max-file` in the example above) must therefore be enclosed in quotes (`"`).
```
Comment (thaJeztah) [inline config/containers/logging/json-file.md]: Why was this removed? Is this because of a bad rebase?

File: config/containers/logging/configure.md [hunk 4]
```diff
@@ -43,32 +44,25 @@ example sets two configurable options on the `json-file` logging driver:
 {
   "log-driver": "json-file",
   "log-opts": {
-    "max-size": "10m",
-    "max-file": "3",
```
Comment (thaJeztah) [inline config/containers/logging/configure.md]: Same here

File: config/containers/logging/configure.md [hunk 5]
```diff
@@ -43,32 +44,25 @@ example sets two configurable options on the `json-file` logging driver:
...
     "labels": "production_status",
     "env": "os,customer"
   }
 }
 ```
 
-> **Note**: `log-opt` configuration options in the `daemon.json` configuration
-> file must be provided as strings. Boolean and numeric values (such as the value
-> for `max-file` in the example above) must therefore be enclosed in quotes (`"`).
```
Comment (thaJeztah) [inline config/containers/logging/configure.md]: And here; looks like a rebase issue / branch out of date

File: config/containers/logging/configure.md [hunk 6]
```diff
@@ -43,32 +44,25 @@ example sets two configurable options on the `json-file` logging driver:
...
 the default output for commands such as `docker inspect <CONTAINER>` is JSON.
 
 To find the current default logging driver for the Docker daemon, run
 `docker info` and search for `Logging Driver`. You can use the following
-command:
+command on Linux, macOS, or PowerShell on Windows:
 
-{% raw %}
 ```bash
-$ docker info --format '{{.LoggingDriver}}'
```
Comment (thaJeztah) [inline config/containers/logging/configure.md]: I think this one was added so that it worked on any platform (not just linux)

File: config/containers/logging/local.md [hunk 7]
```diff
@@ -1,46 +1,53 @@
...
+description: Describes how to use the local logging driver.
+keywords: local, docker, logging, driver
 redirect_from:
 - /engine/reference/logging/local/
 - /engine/admin/logging/local/
-title: local binary file Protobuf logging driver
+title: Local File logging driver
 ---
 
-This `log-driver` writes to `local` binary files using Protobuf [Protocol Buffers](https://en.wikipedia.org/wiki/Protocol_Buffers)
```
Comment (darwin-traver) [inline config/containers/logging/local.md]: Is there a reason for removing the reference to the 3rd party file format? (maybe due to exclusive access comment below?)

**Comment by bermudezmt** [comment]
@shampabh @JustinINevill This is related to the "dual logging" issue. Please prioritize ASAP.

File: config/containers/logging/configure.md (hunk 5 above)
Comment (bermudezmt) [reply config/containers/logging/configure.md]: Yeah, @thaJeztah - this was my quick attempt at producing a `diff` for what was merged into Amberjack vs. what lives in `master`.

File: config/containers/logging/local.md (hunk 7 above)
Comment (bermudezmt) [reply config/containers/logging/local.md]: @darwin-traver I hope we can get some answers soon. :)

**Comment by paigehargrave** [comment]
@anusha-ragunathan Hi Anusha - PTAL - thank you!

File: config/containers/logging/configure.md (hunk 2 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: I deleted this statement.

File: config/containers/logging/configure.md (hunk 4 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Added these back in.

File: config/containers/logging/configure.md (hunk 5 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Wasn't sure if this was needed so left it out. Please let me know if this should be added back in.

File: config/containers/logging/configure.md (hunk 6 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Added original info back in - please let me know if I misunderstood your comment.

File: config/containers/logging/configure.md (hunk 1 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: I added a reference to the new topic (not yet created) so we will need to add the link.

File: config/containers/logging/configure.md (hunk 5 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Crap - just realized that it probably needs to be included. Just added.

File: config/containers/logging/json-file.md (hunk 3 above)
Comment (paigehargrave) [reply config/containers/logging/json-file.md]: Added back in.

File: config/containers/logging/local.md (hunk 7 above)
Comment (paigehargrave) [reply config/containers/logging/local.md]: Anusha has indicated that she will review this.

**Comment by paigehargrave** [comment]
@anusha - PTAL - thank you again for all of your help and input on this!

**Comment by paigehargrave** [comment]
@darwin-traver @thaJeztah @bermudezmt - I added a new topic based on Anusha's input. PTAL. I'm hoping to get this puppy concluded soon. :) Thanks in advance!

File: config/containers/logging/configure.md [hunk 8]
```diff
@@ -157,18 +157,21 @@ see more options.
 
 ## Limitations of logging drivers
 
-Depending on the version of Docker Engine that you are using, the `docker logs` command is only available on the following drivers:
-- `json-file`
-- `journald`
-- `local`
+- Depending on the version of Docker Engine that you are using, the `docker logs` command is only available on the following drivers:
```
Comment (anusha-ragunathan) [inline config/containers/logging/configure.md]: We should specify the exact version of Docker Engine that this limitation exists. Something like "On Docker Engine version XX.YY and earlier,..."

File: config/containers/logging/configure.md [hunk 9]
```diff
@@ -157,18 +157,21 @@ see more options.
...
+    - `json-file`
+    - `journald`
+    - `local`
 
 Refer to [link needed to new topic]() for information about using `docker logs` to read container logs locally for many third party logging solutions, including:
```
Comment (anusha-ragunathan) [inline config/containers/logging/configure.md]: Add link as part of this PR

File: config/containers/logging/configure.md [hunk 10]
```diff
@@ -43,32 +42,30 @@ example sets two configurable options on the `json-file` logging driver:
 {
   "log-driver": "json-file",
   "log-opts": {
-    "max-size": "10m",
+    "max-size": "10m",
```
Comment (anusha-ragunathan) [inline config/containers/logging/configure.md]: is that a trailing tab? If so, please remove it

File: config/containers/logging/configure.md [hunk 11]
```diff
@@ -43,32 +42,30 @@ example sets two configurable options on the `json-file` logging driver:
...
     "labels": "production_status",
     "env": "os,customer"
   }
 }
 ```
 
-> **Note**: `log-opt` configuration options in the `daemon.json` configuration
-> file must be provided as strings. Boolean and numeric values (such as the value
-> for `max-file` in the example above) must therefore be enclosed in quotes (`"`).
+**Note**: `log-opt` configuration options in the `daemon.json` configuration
```
Comment (anusha-ragunathan) [inline config/containers/logging/configure.md]: Trailing tabs again.

File: config/containers/logging/configure.md (hunk 8 above)
Comment (anusha-ragunathan) [reply config/containers/logging/configure.md]: Thinking more, using `docker logs` to read logs for remote logging drivers is a feature in Docker EE Engine only. 

So this should be rephrased as: 
"`docker logs` support for the `local` logging driver was added in docker CE version XX.YY"

File: config/containers/logging/dual-logging.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
+---
+description: Learn how to read container logs locally when using a third party logging solution.
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: I think this entire file belongs to the `docs-private` repo, since reading the logs for remote drivers is a Docker EE engine feature. @thaJeztah : Can you confirm that we didnt port this feature to CE and moby? Thanks.

File: config/containers/logging/dual-logging.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+keywords: docker, logging, driver
+title: Reading container logs locally with remote logging drivers
+---
+
+## Overview 
+
+In previous versions of Docker Enterprise, the jsonfile and journald log drivers supported reading 
+container logs locally using `docker logs`. However, many third party logging drivers had no
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: "container logs locally" should be
"container logs"

File: config/containers/logging/dual-logging.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+- fluentd	
+- awslogs	
+- splunk	
+- etwlogs	
+- gcplogs	
+- Logentries
+
+This created multiple problems, especially with UCP, when attempting to gather log data in an 
+automated and standard way. Log information could only be accessed and viewed through the 
+third-party solution in the format specified by that third-party tool, which often required authentication.
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: remove "which often required authentication"

File: config/containers/logging/dual-logging.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+- etwlogs	
+- gcplogs	
+- Logentries
+
+This created multiple problems, especially with UCP, when attempting to gather log data in an 
+automated and standard way. Log information could only be accessed and viewed through the 
+third-party solution in the format specified by that third-party tool, which often required authentication. 
+
+Starting with Docker Engine Enterprise 18.03.1-ee-1, you can use `docker logs` to read container 
+logs regardless of the configured logging solution or plugin. This capability, sometimes referred to
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: "configured logging solution or plugin" should be
"configured logging driver or plugin"

File: config/containers/logging/dual-logging.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+This created multiple problems, especially with UCP, when attempting to gather log data in an 
+automated and standard way. Log information could only be accessed and viewed through the 
+third-party solution in the format specified by that third-party tool, which often required authentication. 
+
+Starting with Docker Engine Enterprise 18.03.1-ee-1, you can use `docker logs` to read container 
+logs regardless of the configured logging solution or plugin. This capability, sometimes referred to 
+as dual logging, allows you to use `docker logs` to read container logs locally in a consistent format, 
+regardless of the remote log driver used, because the engine is configured to log information to the “local” 
+logging driver. Refer to [Configure the default logging driver](/configure) for additional information.
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: Add actual link to "configure the default logging driver" here.

File: config/containers/logging/dual-logging.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+
+Starting with Docker Engine Enterprise 18.03.1-ee-1, you can use `docker logs` to read container 
+logs regardless of the configured logging solution or plugin. This capability, sometimes referred to 
+as dual logging, allows you to use `docker logs` to read container logs locally in a consistent format, 
+regardless of the remote log driver used, because the engine is configured to log information to the “local” 
+logging driver. Refer to [Configure the default logging driver](/configure) for additional information. 
+
+**Note**: 
+
+- Dual logging is only available if you are using a third-party logging solution that does not
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: Lines 34 and 35 should be removed.

File: config/containers/logging/dual-logging.md [hunk 18]
```diff
@@ -0,0 +1,116 @@
...
+
+- Dual logging is only available if you are using a third-party logging solution that does not 
+support reading logs via 'docker logs'. 
+- Dual logging is only supported for Docker Enterprise, and is enabled by default starting with 
+Engine Enterprise 18.03.1-ee-1.
+
+## Usage
+Dual logging is enabled by default. You must configure either the docker daemon or the container with remote logging driver. 
+
+The following example shows the results of running a `docker logs` command with and wihtout dual logging availability:
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: `wihtout` -> `without`

File: config/containers/logging/dual-logging.md [hunk 19]
```diff
@@ -0,0 +1,116 @@
...
+
+    ```
+    $ docker run -d busybox --name testlog top 
+    ```
+
+- Step 3: Read the container logs
+    ```
+    $ docker logs 7d6ac83a89a0
+    ```
+    The docker logs command was not available for drivers other than json-file and journald and an error was displayed.
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: `The docker logs command was not available for drivers other than json-file and journald` is the actual error. Should it should be part of the markdown that shows the `$docker logs 7d6ac83a89a0` output above.

Comment (anusha-ragunathan) [reply config/containers/logging/dual-logging.md]: the trailing "an error was displayed" should be removed.

File: config/containers/logging/json-file.md [hunk 20]
```diff
@@ -27,21 +23,19 @@ configuring Docker using `daemon.json`, see
 [daemon.json](/engine/reference/commandline/dockerd.md#daemon-configuration-file).
 
 The following example sets the log driver to `json-file` and sets the `max-size`
-and `max-file` options.
+option.
```
Comment (anusha-ragunathan) [inline config/containers/logging/json-file.md]: Why is `max-file` option removed?

File: config/containers/logging/json-file.md [hunk 21]
```diff
@@ -27,21 +23,19 @@ configuring Docker using `daemon.json`, see
...
 
 ```json
 {
   "log-driver": "json-file",
   "log-opts": {
-    "max-size": "10m",
-    "max-file": "3"
```
Comment (anusha-ragunathan) [inline config/containers/logging/json-file.md]: same here. `max-file` still holds true for json-file. it should not be removed

File: config/containers/logging/local.md [hunk 22]
```diff
@@ -50,6 +57,15 @@ The `local` logging driver supports the following logging options:
...
-| `max-size`  | The maximum size of each binary log file before rotation. A positive integer plus a modifier representing the unit of measure (`k`, `m`, or `g`). Defaults to `20m`.                                          | `--log-opt max-size=10m`                 |
-| `max-file`  | The maximum number of binary log files. If rotating the logs creates an excess file, the oldest file is removed. **Only effective when `max-size` is also set.** A positive integer. Defaults to `5`.         | `--log-opt max-file=5`                   |
-| `compress`  | Whether or not the binary files should be compressed. Defaults to `true`                                                                                                                                      | `--log-opt compress=true`              |
+| `max-size`  | The maximum size of the log before it is rolled. A positive integer plus a modifier representing the unit of measure (`k`, `m`, or `g`). Defaults to 20MB.                                          | `--log-opt max-size=10m`                 |
```
Comment (anusha-ragunathan) [inline config/containers/logging/local.md]: "Defaults to 20MB" ---> "Defaults to 20m"

File: config/containers/logging/configure.md (hunk 8 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Reworded - thank you!

File: config/containers/logging/configure.md (hunk 9 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Got it.

File: config/containers/logging/configure.md (hunk 10 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Got it.

File: config/containers/logging/configure.md (hunk 11 above)
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 13 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it - thanks!

File: config/containers/logging/dual-logging.md (hunk 14 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 15 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 16 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 17 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Removed.

File: config/containers/logging/dual-logging.md (hunk 18 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it - thank you!

File: config/containers/logging/dual-logging.md (hunk 19 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it - thank you!

File: config/containers/logging/json-file.md (hunk 20 above)
Comment (paigehargrave) [reply config/containers/logging/json-file.md]: Not sure - added it back in.

File: config/containers/logging/json-file.md (hunk 21 above)
Comment (paigehargrave) [reply config/containers/logging/json-file.md]: Not sure what happened - added this back in.

File: config/containers/logging/local.md (hunk 22 above)
Comment (paigehargrave) [reply config/containers/logging/local.md]: Got it - thank you!

**Comment by paigehargrave** [comment]
@anusha-ragunathan - thank you for the great input and all of your time! @bermudezmt - Can you PTAL for a final sanity check?

File: config/containers/logging/json-file.md [hunk 23]
```diff
@@ -26,22 +22,20 @@ located in `/etc/docker/` on Linux hosts or
...
-and `max-file` options.
+The following example sets the log driver to `json-file` and sets the `max-size` and 'max-file' options.
 
 ```json
 {
   "log-driver": "json-file",
   "log-opts": {
-    "max-size": "10m",
-    "max-file": "3"
+    "max-size": "10m"
```
Comment (anusha-ragunathan) [inline config/containers/logging/json-file.md]: missing a comma at the end.

File: config/containers/logging/dual-logging.md (hunk 12 above)
Comment (anusha-ragunathan) [reply config/containers/logging/dual-logging.md]: @paigehargrave : This file should not be part of the public moby docs. It should be part of the EE engine docs. You should create another PR in the EE repo with this file and link the rest of the content to that link,

Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: @anusha-ragunathan The EE and CE engine information both live in docs.docker.com - they are not separate at this point. I've included a note about support for CE vs. EE, please let me know if more is needed. Thank you!

File: config/containers/logging/json-file.md (hunk 23 above)
Comment (paigehargrave) [reply config/containers/logging/json-file.md]: Got it - thank you!

File: config/containers/logging/dual-logging.md [hunk 24]
```diff
@@ -0,0 +1,114 @@
+---
+description: Learn how to read container logs locally when using a third party logging solution.
+keywords: docker, logging, driver
+title: Reading container logs locally with remote logging drivers
+---
+
+## Overview 
+
+In previous versions of Docker Enterprise, the jsonfile and journald log drivers supported reading
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: "Docker Enterprise" -> "Docker Engine Enterprise"

File: config/containers/logging/dual-logging.md [hunk 25]
```diff
@@ -0,0 +1,114 @@
+---
+description: Learn how to read container logs locally when using a third party logging solution.
+keywords: docker, logging, driver
+title: Reading container logs locally with remote logging drivers
```
Comment (anusha-ragunathan) [inline config/containers/logging/dual-logging.md]: "Reading container logs locally with remote logging drivers" should be 

"Using `docker logs` to read container logs for remote logging drivers"

File: config/containers/logging/dual-logging.md (hunk 12 above)
Comment (anusha-ragunathan) [reply config/containers/logging/dual-logging.md]: Where is the note about support for CE vs EE? I think the beginning of this page should make it obvious that you need a Docker EE Engine for this feature. Maybe have a prerequisite section?

Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: It's in a note in the overview section.

File: config/containers/logging/dual-logging.md (hunk 25 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 24 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Got it.

File: config/containers/logging/dual-logging.md (hunk 12 above)
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: I changed the note to 'Prerequisites'.

**Comment by anusha-ragunathan** [comment]
Looks good. Almost. 

Only pending item that I need confirmation from @thaJeztah : Can you confirm that reading logs using `docker logs` for remote logging driver is a EE engine only feature and not available on CE/moby?

**Comment by thaJeztah** [comment]
Yes, that feature isn't available on Docker CE, only on EE

**Comment by anusha-ragunathan** [comment]
LGTM!

**Comment by paigehargrave** [comment]
@anusha-ragunathan @thaJeztah Thank you! @bermudezmt Just need to get with you for a quick review on this. Thanks!

File: config/containers/logging/configure.md [hunk 26]
```diff
@@ -157,9 +151,25 @@ see more options.
 | [`etwlogs`](etwlogs.md)       | Writes log messages as Event Tracing for Windows (ETW) events. Only available on Windows platforms.           |
 | [`gcplogs`](gcplogs.md)       | Writes log messages to Google Cloud Platform (GCP) Logging.                                                   |
 | [`logentries`](logentries.md) | Writes log messages to Rapid7 Logentries.                                                                     |
+| [`local`](local.md)           | Logs are stored in a custom format designed for minimal overhead.                                             |
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: 'local' entry moved up

File: config/containers/logging/configure.md [hunk 27]
```diff
@@ -51,24 +50,20 @@ example sets two configurable options on the `json-file` logging driver:
...
+command on Linux, macOS, or PowerShell on Windows:
 
-{% raw %}
 ```bash
 $ docker info --format '{{.LoggingDriver}}'
 
-json-file
+Logging Driver: json-file
 ```
-{% endraw %}
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Got it.

File: config/containers/logging/configure.md [hunk 28]
```diff
@@ -51,24 +50,20 @@ example sets two configurable options on the `json-file` logging driver:
...
 
 If you do not specify a logging driver, the default is `json-file`. Thus,
 the default output for commands such as `docker inspect <CONTAINER>` is JSON.
 
 To find the current default logging driver for the Docker daemon, run
 `docker info` and search for `Logging Driver`. You can use the following
-command:
+command on Linux, macOS, or PowerShell on Windows:
 
-{% raw %}
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Got it.

File: config/containers/logging/configure.md [hunk 29]
```diff
@@ -157,9 +151,25 @@ see more options.
...
 ## Limitations of logging drivers
 
-The `docker logs` command is not available for drivers other than `json-file`
-and `journald`.
+- The `docker logs` command, which provides support for the local logging driver, was added in Docker CE version 17.05. The `docker logs` command is only available on the following drivers:
+    - `json-file`
+    - `journald`
+    - `local`
+
+Refer to [Reading logs when using remote logging drivers](/dual-logging) for information about using `docker logs` to read container logs locally for many third party logging solutions, including:
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: Paragraph moved up.

File: config/containers/logging/configure.md [hunk 30]
```diff
@@ -157,9 +151,25 @@ see more options.
...
+    - syslog	
+    - gelf	
+    - fluentd	
+    - awslogs	
+    - splunk	
+    - etwlogs	
+    - gcplogs	
+    - Logentries
 
+- Reading log information requires decompressing rotated log files, which causes a temporary increase in disk usage (until the log entries from the rotated files are read) and an increased CPU usage while decompressing.
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: @anusha-ragunathan - Can you confirm whether this is still the case?

File: config/containers/logging/configure.md [hunk 31]
```diff
@@ -157,9 +151,25 @@ see more options.
...
+    - gelf	
+    - fluentd	
+    - awslogs	
+    - splunk	
+    - etwlogs	
+    - gcplogs	
+    - Logentries
 
+- Reading log information requires decompressing rotated log files, which causes a temporary increase in disk usage (until the log entries from the rotated files are read) and an increased CPU usage while decompressing. 
+- The capacity of the host storage where docker’s data directory resides determines the maximum size of the log file information.
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: @anusha-ragunathan - Can you PTAL?

File: config/containers/logging/dual-logging.md [hunk 32]
```diff
@@ -0,0 +1,114 @@
+---
+description: Learn how to read container logs locally when using a third party logging solution.
+keywords: docker, logging, driver
+title: Using docker logs to read container logs for remote logging drivers
+---
+
+## Overview 
+
+In previous versions of Docker Engine Enterprise, the jsonfile and journald log drivers supported reading
```
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/dual-logging.md]: Added text.

**Comment by paigehargrave** [comment]
@anusha-ragunathan - Two things left for your final review - PTAL. Thank you!

File: config/containers/logging/configure.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (anusha-ragunathan) [reply config/containers/logging/configure.md]: https://github.com/moby/moby/blob/77faf158f5265711dbcbff0ffb855eed2e3b6ccd/daemon/logger/local/local.go#L28-L30 shows that local logs are compressed by default. So reading logs *will* require decompressing rotated log files, which can cause in a spike in cpu and disk usage.

File: config/containers/logging/configure.md (hunk 31 above)
[the remark this answers is not in the document]
Comment (anusha-ragunathan) [reply config/containers/logging/configure.md]: I agree with @thaJeztah. `local` logger has default max file size and max file count. So its not gonna consume all your disk space.

Regarding recommending `local` logging driver, I remember a conversation back during implementation about why having "local" driver as default would break compatibility. @thaJeztah : do you remember why? Wondering if that reason could prevent us from recommending "local" logger.

File: config/containers/logging/configure.md (hunk 30 above)
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: @anusha-ragunathan Thank you! I'm leaving text as is.

File: config/containers/logging/configure.md (hunk 31 above)
[the remark this answers is not in the document]
Comment (paigehargrave) [reply config/containers/logging/configure.md]: @anusha-ragunathan Thank you! @thaJeztah - PTAL - once this is resolved, we can ship this! Thank you both so much again for your input here!
